Samples/excel-data-types-explorer/src/taskpane/taskpane.ts

```diff
@@ -0,0 +1,1023 @@
+          )
+        )
+        .append(
+          $("<tr/>").append(textInputWithLabel("providerTarget", "Provider address: ", "Provider source input box"))
+        )
+        .append($("<tr/>").append(textInputWithLabel("providerLogo", "Logo address: ", "Provider logo input box")))
+    )
+  );
+
+  var referencedValuesLabel = $(`<label for="referencedValues" class="labels">Referenced values: </label>`);
+  var referencedValuesInput = $(
+    `<input class="inputBox displayString" type="text" id="referencedValues" alt="referencedValues text display box" disabled/>`
+  );
+  var referencedValuesClearButton = $("<button/>", {
+    id: "clearReferencedValue",
+    class: "ms-Button ms-Button-label buttons",
+    text: "Clear referencedValues"
+  });
+
+  var referencedValuesDiv = $("<div/>", { id: "referencedValuesDiv" })
+    .append(referencedValuesLabel)
+    .append(referencedValuesInput)
+    .append(referencedValuesClearButton);
+
+  referencedValuesDiv[0].style.display = "none";
+
+  referencedValuesClearButton[0].onclick = function() {
+    referencedValuesInput.val("");
+    referencedValuesDiv[0].style.display = "none";
+  };
+
+  var providerInfo = $("<div/>", { class: "contentPadding" })
+    .append($("<label/>", { class: "labels" }).text("Provider info: "))
+    .append(providerTable);
+
+  var element = $(`<tr id="Entity"/>`).append(
+    $(`<td colspan="2"/>`)
+      .append([label, input])
+      .append(div)
+      .append(providerInfo)
+      .append(referencedValuesDiv)
+      .append(contentLabel)
+      .append(sections)
+  );
+
+  return element;
+}
+
+/** Create the HTML for when boolean, string, or double data types are selected */
+function setSelectedType(selected: JQuery<HTMLElement>[] | JQuery<HTMLElement>) {
+  var element = $(`<div class= "backgroundColorForm solidBorder"/>`);
+  element.append(selected);
+
+  $(".backgroundColorForm").replaceWith(element);
+}
+
+/** Add a new field to the entity fields. */
+function addField(element: HTMLButtonElement) {
+  // Append specificFieldContents to the fieldsDiv.
+  const fieldsDiv = element.parentElement.previousElementSibling;
+  $(fieldsDiv).append(specificFieldContent());
+}
+
+/** Expand the section fields in the entity field. */
+function expandSection(element: HTMLButtonElement) {
+  $(element).replaceWith(
+    `<button class="ms-Button ms-Button-label arrows sectionToggle" onclick="collapseSection(this)" alt="collapse section" title="collapse section">&#x2228</button>`
+  );
+}
+
+/** Collapse the section fields in the entity field. */
+function collapseSection(element: HTMLButtonElement) {
+  $(element).replaceWith(
+    `<button class="ms-Button ms-Button-label arrows sectionToggle" onclick="expandSection(this)" alt="expand section" title="expand section" >&#x2227 </button>`
+  );
+}
+
+/** Assign the inputted data to the active worksheet cell as the appropriate data type. */
+function createValueAsJson(): Excel.CellValue {
```

Comment:
Consider having the addition of the type to the worksheet clear the form as well.

Reply:
Done!